### PR TITLE
feat(backups): get transferred data length

### PR DESCRIPTION
**Pull request by badrAZ**
Fixes vatesfr/xo-web#2096

### CODE REVIEW

File: src/xo-mixins/backups.js [hunk 1]
```diff
@@ -703,7 +708,10 @@ export default class {
     }
 
     // Returns relative path.
```
Comment (julien-f) [inline src/xo-mixins/backups.js]: This comment needs to be moved one line below.

File: src/xo-mixins/backups.js [hunk 2]
```diff
@@ -703,7 +708,10 @@ export default class {
...
-    return `${backupDirectory}/${vdiFilename}`
+    return {
+      path: `${backupDirectory}/${vdiFilename}`,
+      length: dataLength
```
Comment (julien-f) [inline src/xo-mixins/backups.js]: `length` → `size`

File: src/xo-mixins/backups.js [hunk 3]
```diff
@@ -852,7 +864,10 @@ export default class {
     }
 
     // Returns relative path.
```
Comment (julien-f) [inline src/xo-mixins/backups.js]: Move the comment.

File: src/xo-mixins/backups.js [hunk 4]
```diff
@@ -852,7 +864,10 @@ export default class {
...
-    return `${dir}/${backupFormat}`
+    return {
+      path: `${dir}/${backupFormat}`,
+      length: dataLength
```
Comment (julien-f) [inline src/xo-mixins/backups.js]: `length` → `size`.

File: src/xo-mixins/backups.js [hunk 5]
```diff
@@ -692,7 +694,10 @@ export default class {
       stream.on('error', error => targetStream.emit('error', error))
 
       await Promise.all([
-        eventToPromise(stream.pipe(targetStream), 'finish'),
+        eventToPromise(stream.pipe(through2((chunk, enc, cb) => {
+          dataLength += chunk.length
+          cb(null, chunk)
+        })).pipe(targetStream), 'finish'),
```
Comment (julien-f) [inline src/xo-mixins/backups.js]: IMHO it would be nice to abstract this in a module, don't you think?

Comment (badrAZ) [inline src/xo-mixins/backups.js]: Yes, it would be better.

File: src/xo-mixins/backups.js [hunk 6]
```diff
@@ -918,9 +936,16 @@ export default class {
...
+
+    const sizeStream = createSizeStream()
+
+    sourceStream
+      .pipe(sizeStream)
+      .pipe(targetStream)
 
     await promise
+
+    return sizeStream.size
```
Comment (julien-f) [inline src/xo-mixins/backups.js]: Try to be coherent: always return an object :)

**Comment by julien-f** [comment]
@badrAZ Can you rebase and resolve the conflicts?

**Comment by badrAZ** [comment]
Yes, i will do it.

**Pull request merged**